VmAgent.Core.UnitTets/SystemOperationsTests.cs

```diff
@@ -0,0 +1,293 @@
+using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Moq;
+using VmAgent.Core.Interfaces;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace VmAgent.Core.UnitTets
+{
+    [TestClass]
+    public class SystemOperationUnitTest
+    {
+        private string _root = "C:\\";
+        private string _directoryPath;
+        private string _subdirectoryPath;
+        private string _directoryFilePath;
+        private string _subdirectoryFilePath;
+        private string _subdirectoryName = "subdirectory\\";
+        private string _directoryName = "directory\\";
+        private string _directoryFileName = "file";
+        private string _subdirectoryFileName = "subfile";
+        private DirectoryInfo _subDirectoryInfo;
+        private DirectoryInfo _directoryInfo;
+        private SystemOperations _systemOperations;
+        private VmConfiguration _vmConfiguration;
+        private MultiLogger _logger;
+        private Mock<IFileSystemOperations> _mockFileSystemOperations;
+
+        [TestInitialize]
+        public void BeforeEachTest()
+        {
+            _directoryPath = Path.Combine(_root, _directoryName);
+            _directoryInfo = new DirectoryInfo(_directoryPath);
+            _subdirectoryPath = Path.Combine(_directoryPath, _subdirectoryName);
+            _subDirectoryInfo = new DirectoryInfo(_subdirectoryPath);
+            _directoryFilePath = Path.Combine(_directoryPath, _directoryFileName);
+            FileInfo directoryFileInfo = new FileInfo(_directoryFilePath);
+            _subdirectoryFilePath = Path.Combine(_subdirectoryPath, _subdirectoryFileName);
+            FileInfo subDirectoryFileInfo = new FileInfo(_subdirectoryFilePath);
+
+            _mockFileSystemOperations = new Mock<IFileSystemOperations>();
+            _vmConfiguration = new VmConfiguration(56001, "vmid", new VmDirectories("root"), true);
+            _logger = new MultiLogger(NullLogger.Instance, new TelemetryClient(TelemetryConfiguration.CreateDefault()));
+            _systemOperations = new SystemOperations(_vmConfiguration, _logger, _mockFileSystemOperations.Object);
+
+            _mockFileSystemOperations.Setup(x => x.IsDirectory(_root)).Returns(true);
+            _mockFileSystemOperations.Setup(x => x.IsDirectory(_subdirectoryPath)).Returns(true);
+            _mockFileSystemOperations.Setup(x => x.IsDirectory(_directoryPath)).Returns(true);
+            _mockFileSystemOperations.Setup(x => x.IsDirectory(_directoryFilePath)).Returns(false);
+            _mockFileSystemOperations.Setup(x => x.IsDirectory(_subdirectoryFilePath)).Returns(false);
+
+            _mockFileSystemOperations.Setup(x => x.GetFiles(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _root)), It.IsAny<bool>()))
```

Comment:
Very long, consider breaking up into methods with names we can follow. Helps anyone fiving into the code to navigate.